### ISSUE TITLE
Inconsistent event status

### ISSUE THREAD

**Issue by ShaunEgan** [walmartlabs/bigben#3]
When an event is received, the status is sent as `TRIGGERED`. However, when using the `find` endpoint and retrieving this event, the status is `PROCESSED`. Is this expected?

**Comment by sandeepmalik** [comment]
That is correct, when BigBen calls the tenant, there's no knowing in advance that the call will succeed or not (the http API may be down, or kafka producer may time out, etc). Hence it calls with status TRIGGERED. Only if the call succeeds (after any number of configured retries), it marks the event as PROCESSED.

The status TRIGGERED (due to failure of the tenant to respond) is not preserved in db because it can be sort of inferred from the following condition: status == UN_PROCESSED && error != null

Let me know what you think.
